plan/logical_plan_builder.go

```diff
@@ -93,6 +88,7 @@ func (b *planBuilder) buildAggregation(p LogicalPlan, aggFuncList []*ast.Aggrega
 	agg.GroupByItems = gbyItems
 	agg.SetSchema(schema)
 	agg.collectGroupByColumns()
+	agg.SetCorrelated()
```

Comment:
We can't, because we need to decide to use semi-join or not during plan building.